src/storage/wal/wal.cpp

```diff
@@ -160,10 +153,9 @@ void WAL::addNewWALRecordNoLock(WALRecord& walRecord) {
 void WAL::setIsLastRecordCommit() {
     WALIterator walIterator(fileHandle, mtx);
     WALRecord walRecord;
+    KU_ASSERT(walIterator.hasNextRecord());
     if (!walIterator.hasNextRecord()) {
-        logger->info(
-            "Opening an existing WAL file but the file is empty. This should never happen. file: " +
-            fileHandle->getFileInfo()->path);
+        // Opening an existing WAL file but the file is empty. This should never happen.
```

Comment:
I wonder if we should assert for a behaviour that will never happen.

Reply:
I asserted before the check. But kept the code here for now under release mode.